securedrop/tests/test_secure_tempfile.py

```diff
@@ -0,0 +1,85 @@
+# -*- coding: utf-8 -*-
+import os
+import unittest
+
+from gnupg._util import _is_stream
+
+os.environ['SECUREDROP_ENV'] = 'test'
+import config
+import secure_tempfile
+import utils
+
+
+class TestSecureTempfile(unittest.TestCase):
+    def setUp(self):
+        utils.env.setup()
+        self.f = secure_tempfile.SecureTemporaryFile(config.STORE_DIR)
+        self.msg = '410,757,864,530'
+
+    def tearDown(self):
+        utils.env.teardown()
+
+    def test_write_then_read_twice(self):
+        self.f.write(self.msg)
+        self.f.read()
+
+        self.assertEqual(self.f.read(), '')
+
+    def test_read_before_writing(self):
+        with self.assertRaisesRegexp(AssertionError,
+                                     'You must write before reading!'):
+            self.f.read()
+        
+    def test_write_then_read_once(self):
+        self.f.write(self.msg)
+
+        self.assertEqual(self.f.read(), self.msg)
+
+    def test_write_then_read_then_write(self):
+        self.f.write(self.msg)
+        self.f.read()
+
+        with self.assertRaisesRegexp(AssertionError,
+                                     'You cannot write after reading!'):
+            self.f.write('BORN TO DIE')
+
+    def test_read_write_unicode(self):
+        unicode_msg = u'BORN TO DIE'
```

Comment:
This test would probably be better if if `unicode_msg` included some characters that cannot be encoded with the ASCII codec, just to make sure they're not getting somehow mangled or dropped along the way. Something similar to the test string from `test_unicode_reply_with_ansi_env`, perhaps?